File: Old/RemovedNativeApi/Sizer/BoxSizer.cs
#pragma warning disable
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Alternet.Drawing;

namespace NativeApi.Api
{
    //https://docs.wxwidgets.org/3.2/classwx_box_sizer.html
    //The basic idea behind a box sizer is that windows will most often be laid out in
    //rather simple basic geometry, typically in a row or a column or several hierarchies
    //of either.
    public class BoxSizer : Sizer
    {
        public static IntPtr CreateBoxSizer(int orient) => default;

        // Adds non-stretchable space to the main orientation of the sizer only.
        public static IntPtr AddSpacer(IntPtr handle, int size) => default;

        // Implements the calculation of a box sizer's minimal. 
        public static SizeI CalcMin(IntPtr handle) => default;

        // Returns the orientation of the box sizer, either wxVERTICAL or wxHORIZONTAL.
        public static int GetOrientation(IntPtr handle) => default;

        // Sets the orientation of the box sizer, either wxVERTICAL or wxHORIZONTAL. 
        public static void SetOrientation(IntPtr handle, int orient) { }

        public static void RepositionChildren(IntPtr handle, SizeI minSize) { }
    }
}

File: Old/RemovedNativeApi/RemovedFromPal/BoxSizer.cpp
#include "BoxSizer.h"

#include <wx/sizer.h>

namespace Alternet::UI
{
    void* BoxSizer::CreateBoxSizer(int orient)
    {
        return new wxBoxSizer(orient);
    }

    void* BoxSizer::AddSpacer(void* handle, int size)
    {
        return ((wxBoxSizer*)handle)->AddSpacer(size);
    }

    Int32Size BoxSizer::CalcMin(void* handle)
    {
        return ((wxBoxSizer*)handle)->CalcMin();
    }

    int BoxSizer::GetOrientation(void* handle)
    {
        return ((wxBoxSizer*)handle)->GetOrientation();
    }

    void BoxSizer::SetOrientation(void* handle, int orient)
    {
        ((wxBoxSizer*)handle)->SetOrientation(orient);
    }

    void BoxSizer::RepositionChildren(void* handle, const Int32Size& minSize)
    {
        ((wxBoxSizer*)handle)->RepositionChildren(minSize);
    }

    BoxSizer::BoxSizer()
    {
    }

    BoxSizer::~BoxSizer()
    {
    }
}
